src/strategies/fight-club/index.ts

```diff
@@ -0,0 +1,104 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { Multicaller } from '../../utils';
+
+export const author = 'MantisClone';
+export const version = '0.1.0';
+
+const abi = [
+  'function balanceOf(address account) external view returns (uint256)',
+  'function balanceOf(address account, uint256 id) external view returns (uint256)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+
+  Object.entries(options.gloveAddresses).forEach(([gloveAddress]) => {
+    addresses.forEach((address: string) => {
+      multi.call(
+        `${address}.gloves.${gloveAddress}`,
+        gloveAddress,
+        'balanceOf(address)',
+        [address]
+      );
+    });
+  });
+
+  Object.entries(options.weightClassIds).forEach(([weightClassId]) => {
```

Comment:
Hi @MantisClone Sorry for not noticing this before, but I feel we should add a limit to `gloveAddresses` and `weightClassIds` if someone passes unlimited entries, there could be memory issues. wyt?

Reply:
I'd be okay with adding a limit. I've decided to arbitrarily limit it to 10 glove colors and 10 weight classes. 

If I had unlimited time, I'd run a test to determine the actual limits. But in practice, I know there will only be 4 glove colors and 4 or 5 weight classes. So setting an arbitrary limit is the more pragmatic approach at this time.